modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -26,6 +26,33 @@
     @defgroup gapi_transform Graph API: Image and channel composition functions
 @}
  */
+
+namespace {
+// Checks if the passed mat is a set of n-dimentional points of the given depth
+// Returns (quantity, elemSize)
+std::tuple<int, int> checkVector(const int chan, const cv::Size &size, const int depth,
+                                 const int ddepth)
```

Comment:
I already saw similar functionality recently.

---

@dmatveev There is a lot of code going into the public headers:
- this may break bindings generators because they parse public headers
- this slowdowns compilation of OpenCV itself and user apps

There is strong recommendation to keep minimal amount of code in public headers.

Reply:
In `imgproc.hpp`. @OrestChura, can you merge `checkVector` and `isPointsVector`?

Reply:
Thank you, Alexander, Maxim!
I made attempt to merge them and share the generalized function between `gapi/core.hpp` & `gapi/imgproc.hpp`. Put it in `gmat.hpp`/`gmat.cpp`